wgpu-hal/src/metal/command.rs

```diff
@@ -1054,3 +1055,17 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
         encoder.dispatch_thread_groups_indirect(&buffer.raw, offset, self.state.raw_wg_size);
     }
 }
+
+impl Drop for super::CommandEncoder {
+    fn drop(&mut self) {
+        // Metal asserts when a MTLCommandEncoder is deallocated without a call
+        // to endEncoding. This isn't documented at
+        // https://developer.apple.com/documentation/metal/mtlcommandencoder?language=objc
+        // but it manifests as a crash with the message 'Command encoder released without
+        // endEncoding'. To prevent this, we explicitiy call discard_encoding, which
+        // calls end_encoding on any still-held metal::CommandEncoders.
```

Comment:
I understand your use of the verb `asserts` here means "to raise an assert", but reads initially as "metal guarantees" 

```suggestion
        // Metal requires that a MTLCommandEncoder is ended before being deallocated.
        // This isn't documented at
        // https://developer.apple.com/documentation/metal/mtlcommandencoder?language=objc
        // but it manifests as a crash with the message 'Command encoder released without
        // endEncoding'. To prevent this, we explicitiy call discard_encoding, which
        // calls end_encoding on any still-held `metal::CommandEncoders`.
```